### DISCUSSION TITLE
In Dexie cloud, how to disable syncing?

### DISCUSSION THREAD

**Discussion by porkopek** [dexie/Dexie.js#2030] [Q&A] [answered]
I'm working on an app with a large amount of data stored on the desktop browser (hundreds of MB). Sometimes, when the user has to sync too much data, syncing can freeze the operations to read the db.
How can I disable the syncing part, so I can make a button to toggle sync?
I saw I can do this:
```JS
  disableEagerSync: true,
  disableWebSocket: true
  ```
  
  but I think this doesn't stop dexieCloud addon to try to sync, because I see in the console there are messages from `syncIfPossible` or `LocalSyncWorker`
  
  I also don't have enabled `tryUseServiceWorker`
  
  If I don't do the part of the `db.cloud.configure`, then it reads from another local db, not that with the url of cloud db in the name.
  I mean, if my db is TGM, and the url is `'https://XXX.dexie.cloud`, with db.cloud.configure active, it reads from `TGM-XXX`, but without it, reads from `TGM` and if I add the XXX part when I instantiate new Dexie, the it will sync again.
  
  So, is there an option to read from **local** db with the cloud db in the name, with no syncing?

**Comment by dfahlander** [comment, marked as answer]
There's a similar request #1828 and I had a suggestion to allow stopping and starting of sync. Probably also would need the initial sync mode to be configurable.

The special naming of the DB can be disabled by turning off the [nameSuffix](https://dexie.org/cloud/docs/db.cloud.configure()#namesuffix) config option.

**Reply by porkopek** [reply]
Thanks, David!

I think is a wanted thing to have, because in large datasets maybe you don't want to sync if your connection is slow at that time. Or just want to make the work locally (no blocking sync) and then sync manually at the end.

I didn't get the Dexie service-worker to work (I'm in a React app, with Vite and PWA plugin), but I'll try again. Maybe this offloads the work from UI thread to the service worker and thus not blocking the UI?

Thanks for the tip on the nameSuffix!

**Reply by dfahlander** [reply]
Another thing I want to mention regarding large a amount of sync data is the support for dividing large sync requests into chunks to prohibit ram consumption on both client and server. This is also planned in the Q3 release.

Currently all fokus is going to Yjs support in Dexie Cloud but these features are highly prioritized after that.

**Reply by porkopek** [reply]
I saw the issue where you were talking about that. It will be nice to have.
I also saw you think large amount of data is not Dexie's target, but I think in my case it might work.
Thanks for this great library and for your time!
